MetaMask/metamask-extension
# Add desktop build type

## Explanation

Adds the `desktop` build type to isolate any future code required to connect to the [MetaMask Desktop](https://github.com/MetaMask/metamask-desktop) application.

As the desktop application will support snaps, this build type will require all the existing `flask` code fences and so basic build type inheritance has also been implemented to avoid adding the `desktop` build type to every existing and future `flask` code fence.

Additional CircleCI jobs have also been added to verify the desktop build type in the pipeline.

## Manual Testing Steps

`yarn start --build-type desktop`

As no desktop support code has been added yet, any manual testing should simply verify the build appears and functions identically to the Flask build.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
